### PR TITLE
Type annotations

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- convert python 2 type annotations to python 3

**Status (please check what you already did)**:
- [x] made PR ready for code review

### CODE REVIEW

File: rasa_nlu/config.py [hunk 1]
```diff
@@ -24,12 +22,11 @@
...
     """Raised if an invalid configuration is encountered."""
 
-    def __init__(self, message):
-        # type: (Text) -> None
+    def __init__(self, message: Text) -> None:
         super(InvalidConfigError, self).__init__(message)
 
 
-def load(filename=None, **kwargs):
+def load(filename: Text = None, **kwargs: Any) -> 'RasaNLUModelConfig':
```
Comment (ricwo) [inline rasa_nlu/config.py]:
```suggestion
def load(filename: Optional[Text] = None, **kwargs: Any) -> 'RasaNLUModelConfig':
```

File: rasa_nlu/data_router.py [hunk 2]
```diff
@@ -384,8 +388,7 @@ def evaluate(self, data, project=None, model=None):
                 errors_filename=None
         )
 
-    def unload_model(self, project, model):
-        # type: (Text, Text) -> Dict[Text]
+    def unload_model(self, project: Text, model: Text) -> Dict[Text, Any]:
```
Comment (ricwo) [inline rasa_nlu/data_router.py]:
```suggestion
    def unload_model(self, project: Optional[Text], model: Text) -> Dict[Text, Any]:
```

File: rasa_nlu/classifiers/sklearn_intent_classifier.py [hunk 3]
```diff
@@ -70,11 +60,10 @@ class SklearnIntentClassifier(Component):
...
 
     def __init__(self,
-                 component_config=None,  # type: Dict[Text, Any]
-                 clf=None,  # type: sklearn.model_selection.GridSearchCV
-                 le=None  # type: sklearn.preprocessing.LabelEncoder
-                 ):
-        # type: (...) -> None
+                 component_config: Dict[Text, Any] = None,
+                 clf: 'sklearn.model_selection.GridSearchCV' = None,
+                 le: 'sklearn.preprocessing.LabelEncoder' = None
```
Comment (ricwo) [inline rasa_nlu/classifiers/sklearn_intent_classifier.py]:
```suggestion
                 le: Optional['sklearn.preprocessing.LabelEncoder'] = None
```

File: rasa_nlu/components.py [hunk 4]
```diff
@@ -375,8 +362,9 @@ def __get_cached_component(self, component_name, model_metadata):
         else:
             return None, cache_key
 
-    def __add_to_cache(self, component, cache_key):
-        # type: (Component, Text) -> None
+    def __add_to_cache(self,
+                       component: Component,
+                       cache_key: Text) -> None:
```
Comment (ricwo) [inline rasa_nlu/components.py]:
```suggestion
                       cache_key: Optional[Text]) -> None:
```

File: rasa_nlu/extractors/crf_entity_extractor.py [hunk 5]
```diff
@@ -76,8 +73,9 @@ class CRFEntityExtractor(EntityExtractor):
         'pattern': lambda doc: doc[3],
     }
 
-    def __init__(self, component_config=None, ent_tagger=None):
-        # type: (sklearn_crfsuite.CRF, Dict[Text, Any]) -> None
+    def __init__(self,
+                 component_config: 'sklearn_crfsuite.CRF' = None,
+                 ent_tagger: Dict[Text, Any] = None) -> None:
```
Comment (ricwo) [inline rasa_nlu/extractors/crf_entity_extractor.py]:
```suggestion
                 ent_tagger: Optional[Dict[Text, Any]] = None) -> None:
```

File: rasa_nlu/model.py [hunk 6]
```diff
@@ -324,15 +325,17 @@ def create(model_metadata,  # type: Metadata
 
         return Interpreter(pipeline, context, model_metadata)
 
-    def __init__(self, pipeline, context, model_metadata=None):
-        # type: (List[Component], Dict[Text, Any], Optional[Metadata]) -> None
+    def __init__(self,
+                 pipeline: List[Component],
+                 context: Dict[Text, Any],
```
Comment (ricwo) [inline rasa_nlu/model.py]:
```suggestion
                 context: Optional[Dict[Text, Any]],
```

**Comment by tmbo** [comment]
Thanks a lot :rocket:

**Pull request merged**